runtime/pack/dist/opt/termdebug/plugin/termdebug.vim

```diff
@@ -272,45 +333,43 @@ enddef
 
 # Use when debugger didn't start or ended.
 def CloseBuffers()
-  exe $'bwipe! {ptybuf}'
-  exe $'bwipe! {commbuf}'
-  if asmbuf > 0 && bufexists(asmbuf)
-    exe $'bwipe! {asmbuf}'
+  exe $'bwipe! {ptybufnr}'
+  exe $'bwipe! {commbufnr}'
+  if asmbufnr > 0 && bufexists(asmbufnr)
+    exe $'bwipe! {asmbufnr}'
   endif
-  if varbuf > 0 && bufexists(varbuf)
-    exe $'bwipe! {varbuf}'
+  if varbufnr > 0 && bufexists(varbufnr)
+    exe $'bwipe! {varbufnr}'
   endif
   running = 0
   gdbwin = 0
 enddef
 
 # IsGdbRunning(): bool may be a better name?
-def CheckGdbRunning(): string
-  var gdbproc = term_getjob(gdbbuf)
-  var gdbproc_status = 'unknown'
-  if type(gdbproc) == v:t_job
-    gdbproc_status = job_status(gdbproc)
-  endif
-  if gdbproc == v:null || gdbproc_status !=# 'run'
-    var cmd_name = string(GetCommand()[0])
+def IsGdbStarted(): bool
+  # CHECKME: check this implementation
+  var gdbproc_status = job_status(term_getjob(gdbbufnr))
```

Comment:
Is the job guaranteed to have been started? 
If not, need an extra test
```
var gdbproc = term_getjob(gdbbufnr)
var gdbproc_status = gdbproc == null ? '' : job_status(gdbproc)
if gdbproc_status !=# 'run'
```

because `job_status` does not accept a `null`, it does accept a `null_job`; another expression of problematic null handling.
```
job_status(null_job)   # WORKS
job_status(null)       # FAIL E1218: Job Required for argument 1
```

Reply:
I remember that I tested it manually and I ended up with that convoluted implementation. That is why I added the CHECKME comment. If you have time, could you double check, please? Four eyes are better than two sometimes. 

I saw that you opened: https://github.com/vim/vim/pull/14983 
Feel free to adjust this function based on such a change.

Reply:
> I saw that you opened: https://github.com/vim/vim/pull/14983

I closed that in favor of #14984 which is simpler and seems to make more sense.

With either of those fixes,
```vim
var gdbproc_status = job_status(term_getjob(gdbbufnr))
```
works whether or not `gdbbufnr` has a job associated with it.

Reply:
#14984 is merged.

Reply:
Perfect! Thanks for the PR!